airbyte-integrations/connectors/destination-kvdb/integration_tests/integration_test.py

```diff
@@ -0,0 +1,165 @@
+    yield
+    client = KvDbClient(**config)
+    client.delete(list(client.list_keys()))
+
+
+@pytest.fixture(name="client")
+def client_fixture(config) -> KvDbClient:
+    return KvDbClient(**config)
+
+
+def test_check_valid_config(config: Mapping):
+    outcome = DestinationKvdb().check(AirbyteLogger(), config)
+    assert outcome.status == Status.SUCCEEDED
+
+
+def test_check_invalid_config():
+    outcome = DestinationKvdb().check(AirbyteLogger(), {"bucket_id": "not_a_real_id"})
+    assert outcome.status == Status.FAILED
+
+
+def _state(data: Dict[str, Any]) -> AirbyteMessage:
+    return AirbyteMessage(type=Type.STATE, state=AirbyteStateMessage(data=data))
+
+
+def _record(stream: str, str_value: str, int_value: int) -> AirbyteMessage:
+    return AirbyteMessage(
+        type=Type.RECORD, record=AirbyteRecordMessage(stream=stream, data={"str_col": str_value, "int_col": int_value}, emitted_at=0)
+    )
+
+
+def retrieve_all_records(client: KvDbClient) -> List[AirbyteRecordMessage]:
+    """retrieves and formats all records in kvdb as Airbyte messages"""
+    all_records = client.list_keys(list_values=True)
+    out = []
+    for record in all_records:
+        key = record[0]
+        stream = key.split("__ab__")[0]
+        value = record[1]
+        out.append(_record(stream, value["str_col"], value["int_col"]))
+    return out
+
+
+def test_write(config: Mapping, configured_catalog: ConfiguredAirbyteCatalog, client: KvDbClient):
+    """
+    This test verifies that:
+        1. writing a stream in "overwrite" mode overwrites any existing data for that stream
+        2. writing a stream in "append" mode appends new records without deleting the old ones
+        3. The correct state message is output by the connector at the end of the sync
+    """
+    append_stream, overwrite_stream = configured_catalog.streams[0].stream.name, configured_catalog.streams[1].stream.name
+    first_state_message = _state({"state": "1"})
+    first_record_chunk = [
+        # make 5 records in each stream followed by state twice
+        *[_record(append_stream, str(i), i) for i in range(5)],
+        *[_record(overwrite_stream, str(i), i) for i in range(5)],
+    ]
+
+    second_state_message = _state({"state": "2"})
+    second_record_chunk = [
+        *[_record(append_stream, str(i), i) for i in range(5, 10)],
+        *[_record(overwrite_stream, str(i), i) for i in range(5, 10)],
+    ]
+
+    destination = DestinationKvdb()
+
+    expected_states = [first_state_message, second_state_message]
+    output_states = list(
+        destination.write(
+            config, configured_catalog, [*first_record_chunk, first_state_message, *second_record_chunk, second_state_message]
+        )
+    )
+    assert expected_states == output_states, "Checkpoint state messages were expected from the destination"
+
+    expected_records = [*[_record(append_stream, str(i), i) for i in range(10)], *[_record(overwrite_stream, str(i), i) for i in range(10)]]
```

Comment:
just for readability, this could be: 
`expected_records = [a for a in b] + [x for x in y]`

Reply:
good call, done